src/priv.jl

```diff
@@ -67,90 +78,89 @@ words this returns the binary exponent of `x` so that
 
 where `significand ∈ [1, 2)`.
 """
-@inline function ilogbk(d::T) where {T<:Union{Float32,Float64}}
-    m = d < T(2)^-threshold_exponent(T)
-    d = ifelse(m, d * T(2)^threshold_exponent(T), d)
+@inline function ilogbk(d::FloatType)
+    T = eltype(d)
+    m = d < (T(2)^-threshold_exponent(T))
+    d = vifelse(m, d * T(2)^threshold_exponent(T), d)
     q = float2integer(d) & exponent_raw_max(T)
-    q = ifelse(m, q - (threshold_exponent(T) + exponent_bias(T)), q - exponent_bias(T))
+    q = vifelse(m, q - (threshold_exponent(T) + exponent_bias(T)), q - exponent_bias(T))
 end
 
 # similar to ilogbk, but argument has to be a normalized float value
-@inline function ilogb2k(d::T) where {T<:Union{Float32,Float64}}
-    (float2integer(d) & exponent_raw_max(T)) - exponent_bias(T)
+@inline function ilogb2k(d::FloatType)
+    T = eltype(d)
+    I = EquivalentInteger(T)
+    (float2integer(d) & I(exponent_raw_max(T))) - I(exponent_bias(T))
 end
 
+const c20d =  1.06298484191448746607415e-05
```

Comment:
Why remove the let block? They prevent polluting the global namespace with `const`